src/components/ImageViewerContainer.ts

```diff
@@ -0,0 +1,138 @@
+import { UrlHelper } from "../UrlHelper";
+
+interface WrapperProps {
+    class: string;
+    mxObject?: mendix.lib.MxObject;
+    style: string;
+    readOnly: boolean;
+}
+
+interface ImageViewerContainerProps extends WrapperProps {
+    source: DataSource;
+    imageAttribute: string;
+    urlStatic: string;
+    imageStatic: string;
+    width: number;
+    widthUnit: Units;
+    height: number;
+    heightUnit: Units;
+}
+
+interface ImageViewerContainerState {
+    alertMessage?: string;
+    imageUrl: string;
+    isLoading?: boolean;
+}
+
+type DataSource = "systemImage" | "urlAttribute" | "staticUrl" | "staticImage";
+type Units = "auto" | "pixels" | "percentage";
+
+class ImageViewerContainer extends Component<ImageViewerContainerProps, ImageViewerContainerState> {
+    private subscriptionHandle: number;
+    private attributeCallback: (mxObject: mendix.lib.MxObject) => () => void;
+
+    constructor(props: ImageViewerContainerProps) {
+        super(props);
+
+        const alertMessage = ImageViewerContainer.validateProps(props);
+        this.state = {
+            alertMessage,
+            imageUrl: "",
+            isLoading: true
+        };
+        this.attributeCallback = mxObject => () => this.getImageUrl(mxObject);
+    }
+
+    render() {
+        if (this.state.alertMessage) {
+            return createElement(Alert, { message: this.state.alertMessage });
+        }
+        if (this.state.isLoading) {
+            return DOM.div(null,
+                DOM.i({ className: "glyphicon glyphicon-cog glyph-spin" }),
+                DOM.span(null, " Loading ...")
+            );
+        }
+
+        return createElement(ImageViewer, {
+            height: this.props.height,
+            heightUnits: this.props.heightUnit,
+            imageurl: this.state.imageUrl,
+            width: this.props.width,
+            widthUnits: this.props.widthUnit
+
+        });
+    }
+
+    componentWillReceiveProps(newProps: ImageViewerContainerProps) {
+        this.resetSubscriptions(newProps.mxObject);
+        this.setState({ isLoading: true });
+        this.getImageUrl(newProps.mxObject);
+    }
+
+    componentWillUnmount() {
+        if (this.subscriptionHandle) window.mx.data.unsubscribe(this.subscriptionHandle);
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        if (this.subscriptionHandle) window.mx.data.unsubscribe(this.subscriptionHandle);
+
+        if (mxObject) {
+            this.subscriptionHandle = window.mx.data.subscribe({
+                callback: this.attributeCallback(mxObject),
+                guid: mxObject.getGuid()
+            });
+        }
+    }
+
+    public static validateProps(props: ImageViewerContainerProps): string {
+        let message = "";
+        if (props.source === "urlAttribute" && !props.imageAttribute) {
+            message = "Configuration error: for data source Dynamic URL; Dynaic URL attribute is required";
```

Comment:
You are still using imageAttribute? Thought we agreed it's not required. As far as I can see, urlAttribute is only being used for conditional checks in this file.

Typo here... Dynaic.